src/Clash/Cores/Etherbone/WishboneMaster.hs

```diff
@@ -0,0 +1,146 @@
+      (WaitForOp _, Df.Data i) -> (wbPkt i) { strobe=True,  busCycle=True }
+      (Busy, Df.NoData)        -> error "No input data in Busy state, this should be impossible!"
+      (Busy, Df.Data i)        -> (wbPkt i) { strobe=True,  busCycle=True }
+      (WaitForAck c _ _ _, _)  -> wbEmpty   { strobe=False, busCycle=c }
+    wbPkt WishboneOperation{..} = WishboneM2S
+      { addr                = _opAddr
+      , writeData           = fromMaybe undefined _opDat
+      , busSelect           = _opSel
+      , lock                = False
+      , busCycle            = undefined
+      , strobe              = undefined
+      , writeEnable         = isJust _opDat
+      , cycleTypeIdentifier = Classic
+      , burstTypeExtension  = LinearBurst
+      }
+    wbEmpty = emptyWishboneM2S
+
+    fsm
+      :: WishboneMasterState dat  -- state
+      -> Df.Data (WishboneOperation addrWidth selWidth dat)  -- iFwd
+      -> Bool                     -- oBwd
+      -> WishboneMasterState dat  -- nextState
+    fsm st@WaitForOp{} Df.NoData _ = st
+    fsm WaitForOp{} (Df.Data WishboneOperation{..}) _
+      | _opAbort  = WaitForOp False
+      | otherwise = Busy
+    fsm Busy{} Df.NoData _ = error "Sender did not keep Df channel constant!" 
+    fsm Busy{} (Df.Data x) _
+      | wbTerm    = WaitForAck (not $ _opDropCyc x) (dat $ _opDat x) (_opEOR x) (_opEOP x)
+      | otherwise = Busy
+      where
+        dat Nothing  = Just $ readData wbBwd
+        dat (Just _) = Nothing
+    fsm st@WaitForAck{} _ False = st
+    fsm WaitForAck{..} _ True = WaitForOp _wbmCyc
+
+
+-- | Transact operations on the wishbone bus.
+--
+-- This is a Wishbone Classic implementation, meaning that the circuit will give
+-- backpressure while the wishbone bus is busy. Once the wishbone bus has set a
+-- termination signal, the circuit can receive a new operation.
+-- Only once the resulting data has been acknowledged by the @RecordBuillder@
+-- will this circuit start a new operation.
+--
+-- The @busCycle@ is controlled by the @_dropCyc@ bool in the incoming
+-- operation. If this is set, the @CYC@ line is dropped in the @WaitForAck@
+-- state and kept low until a new operation arives. 
+wishboneMasterC
+  :: forall dom addrWidth dat selWidth .
+  ( HiddenClock dom
+  , KnownNat addrWidth
+  , BitPack dat
+  , selWidth ~ ByteSize dat
+  , NFDataX dat
+  , Show dat
+  )
+  => Reset dom
+  -> Circuit (Df.Df dom (WishboneOperation addrWidth selWidth dat))
+             ( Df.Df dom (WishboneResult dat)
+             , Wishbone dom Standard addrWidth dat
+             , CSignal dom (Maybe Bit)
+             )
+wishboneMasterC rst = Circuit $ B.second unbundle . fsm . B.second bundle
+  where
+    fsm inp = withEnable enableGen $ withReset rst mealyB wishboneMasterT (WaitForOp False) (rst', bundle inp)
+    rst' = unsafeToActiveHigh rst
+
+    -- cnt = withReset rst $ withEnable enableGen register (0 :: Unsigned 16) (cnt+1)
+{-# OPAQUE wishboneMasterC #-}
```

Comment:
This was to ensure the invariant 'A component may not assert an acknowledgment while its reset is asserted'. Though, I missed that the mealy machine is in reset. Pretty sure that `forceResetSanity` can't be used because the `Wishbone` protocol is used.

However, this this invariant is probably irrelevant because the whole etherbone circuit is using the same reset. So no data can be sent / lost while reset is asserted.